cps-global-lib/README.md

```diff
@@ -59,28 +65,133 @@ Thereafter it should suffice to run:
 At the base level, any valid Groovy code is OK. So you can define data structures,
 utility functions, and etc., like this:
 
-    $ cat src/org/foo/Point.groovy
-    package org.foo;
+```groovy
+// src/org/foo/Point.groovy
+package org.foo;
 
-    // point in 3D space
-    class Point {
-      float x,y,z;
-    }
+// point in 3D space
+class Point {
+  float x,y,z;
+}
+```
 
 However classes written like this cannot call step functions like `sh` or `git`.
 More often than not, what you want to define is a series of functions that in turn invoke
 other workflow step functions. You can do this by not explicitly defining the enclosing class,
 just like your main workflow script itself:
 
-    $ cat src/org/foo/Zot.groovy
-    package org.foo;
+```groovy
+// src/org/foo/Zot.groovy
+package org.foo;
 
-    def checkOutFrom(repo) {
-      git url: "git@github.com:jenkinsci/${repo}"
-    }
+def checkOutFrom(repo) {
+  git url: "git@github.com:jenkinsci/${repo}"
+}
+```
 
 You can then call such function from your main workflow script like this:
 
-    def z = new org.foo.Zot()
-    z.checkOutFrom(repo)
+```groovy
+def z = new org.foo.Zot()
+z.checkOutFrom(repo)
+```
+
+### Defining global functions
```

Comment:
Hmm, for me the syntax simplification for the generic step is a problem on its own. You are right that this could be a workaround while we solve that, but not sure if we want to advertise it in README.